Mathlib/Algebra/Order/BigOperators/Ring/Finset.lean

```diff
@@ -208,6 +208,40 @@ theorem sq_sum_div_le_sum_sq_div [LinearOrderedSemifield R] [ExistsAddOfLE R] (s
   rw [div_mul_cancel₀]
   exact (hg i hi).ne'
 
+/-- Weighted **AM-HM inequality**: The weighted harmonic mean is less than or equal to the
+arithmetic mean. -/
+theorem inv_sum_div_le_sum_mul [LinearOrderedSemifield R] [ExistsAddOfLE R] (s : Finset ι)
+    {w f : ι → R} (hw : ∑ i ∈ s, w i = 1) (hw' : ∀ i ∈ s, 0 < w i) (hf : ∀ i ∈ s, 0 < f i) :
+    (∑ i ∈ s, w i / f i)⁻¹ ≤ ∑ i ∈ s, w i * f i := by
```

Comment:
Any term with `w i = 0` or `f i = 0` disappears from both the LHS and RHS, so this should still be true
```suggestion
    {w f : ι → R} (hw : ∑ i ∈ s, w i = 1) (hw' : ∀ i ∈ s, 0 ≤ w i) (hf : ∀ i ∈ s, 0 ≤ f i) :
    (∑ i ∈ s, w i / f i)⁻¹ ≤ ∑ i ∈ s, w i * f i := by
```

Reply:
Removing terms `f i` from the sum changes the "effective" sum of the `w i`, which makes the inequality not work. For instance, If `w = ![1/2, 1/2]` and `f = v![1, 1]`, the inequality reads `2 ≤ 1 / 2` which is false.

I think `w i = 0` shouldn't be an issue though.

Reply:
Yeah, good point